NineChronicles.Headless/ActionEvaluationPublisher.cs

```diff
@@ -72,6 +73,18 @@ public class ActionEvaluationPublisher : BackgroundService
                 "ninechronicles_rpc_clients_count",
                 () => this.GetClients().Count,
                 description: "Number of RPC clients connected.");
+            meter.CreateObservableGauge(
+                "ninechronicles_rpc_clients_mobile_count",
+                () => this.GetClientsCountByDevice("mobile"),
+                description: "Number of mobile RPC clients connected.");
+            meter.CreateObservableGauge(
+                "ninechronicles_rpc_clients_pc_count",
+                () => this.GetClientsCountByDevice("pc"),
+                description: "Number of pc RPC clients connected.");
+            meter.CreateObservableGauge(
+                "ninechronicles_rpc_clients_other_count",
+                () => this.GetClientsCountByDevice("other"),
+                description: "Number of other RPC clients connected.");
```

Comment:
How about using `tags` (`labels`) feature instead assigning all metrics?


With tags, it can be presented like the below metrics:

```
process_runtime_dotnet_gc_heap_size_bytes{generation="gen0"} 4414816 1691449546325
process_runtime_dotnet_gc_heap_size_bytes{generation="gen1"} 21489360 1691449546325
process_runtime_dotnet_gc_heap_size_bytes{generation="gen2"} 240 1691449546325
process_runtime_dotnet_gc_heap_size_bytes{generation="loh"} 29081808 1691449546325
process_runtime_dotnet_gc_heap_size_bytes{generation="poh"} 179128 1691449546325
```

In this case, we can use [`CreateObservableGauge(string, Func<IEnumerable<Measuarement<T>>>, string?, string?)`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.meter.createobservablegauge?view=net-7.0#system-diagnostics-metrics-meter-createobservablegauge-1(system-string-system-func((system-collections-generic-ienumerable((system-diagnostics-metrics-measurement((-0))))))-system-string-system-string)) now.

It can be like:

```csharp
var meter = new Meter("NineChronicles");
meter.CreateObservableGauge(
    "ninechronicles_rpc_clients_count",
    () => new []
    {
        new Measurement<int>(this.GetClientsCountByDevice("mobile"), new[] { new KeyValuePair<string, object?>("device", "mobile") }),   
        new Measurement<int>(this.GetClientsCountByDevice("pc"), new[] { new KeyValuePair<string, object?>("device", "pc") }),   
        new Measurement<int>(this.GetClientsCountByDevice("other"), new[] { new KeyValuePair<string, object?>("device", "other") }),   
    },
    description: "Number of RPC clients connected.");
```

Or with LINQ:

```csharp
var meter = new Meter("NineChronicles");
meter.CreateObservableGauge(
    "ninechronicles_rpc_clients_count",
    () => _clientsByDevice.GroupBy(pair => pair.Value)
        .Select(group => new Measurement<int>(group.Count(),
            new[] { new KeyValuePair<string, object?>("device", group.Key) })),
    description: "Number of RPC clients connected.");
```

***LINQ version will not return anything as metrics if there are no clients.***

Reply:
I'll make adjustments on this. Thank you for the feedback!